src/lib/src/service/persistence/conflict.ts

```diff
@@ -0,0 +1,9 @@
+// TODO remove this file (which was copied over from test-editor-web) and refactor its usages out into test-editor-web
+
+export class Conflict {
+  constructor(readonly message: string, readonly backupFilePath?: string) { }
+}
+
+export function isConflict(conflict: Conflict | {}): conflict is Conflict {
```

Comment:
In case of no conflict, isn't the result a 'string' in that case? can 'string' be used here? the type predicate would then make sure that in the 'else' case the type is known to be a 'string' ...

Reply:
Ah, yes, copy&paste error :disappointed: 
In test-editor-web, the alternative actually is `{}`…